### PR TITLE
Adding WSO2 Identity Server Analytics Solution

### PR DESCRIPTION
**Pull request by ThisaruGuruge**
## Purpose
Adding widgets in WSO2 Identity Server Analytics Dashboard.

### CODE REVIEW

File: components/sp-solutions/org.wso2.carbon.sp.solutions.is.analytics/widgets/DateTimeRangePicker/src/styles/main.css [hunk 1]
```diff
@@ -0,0 +1,15 @@
+.perUnderline:before {
```
Comment (dilini-muthumala) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.is.analytics/widgets/DateTimeRangePicker/src/styles/main.css]: Please add license header.

File: components/sp-solutions/org.wso2.carbon.sp.solutions.is.analytics/widgets/DateTimeRangePicker/webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
```
Comment (dilini-muthumala) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.is.analytics/widgets/DateTimeRangePicker/webpack.config.js]: Fix year.

File: components/sp-solutions/org.wso2.carbon.sp.solutions.is.analytics/widgets/IsAnalyticsAreaChart/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
+const path = require('path');
```
Comment (dilini-muthumala) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.is.analytics/widgets/IsAnalyticsAreaChart/webpack.config.js]: Please add license header.

File: components/sp-solutions/org.wso2.carbon.sp.solutions.is.analytics/widgets/IsAnalyticsHorizontalBarChart/src/IsAnalyticsHorizontalBarChart_bk.jsx [hunk 4]
```diff
@@ -0,0 +1,673 @@
+import React, {Component} from "react";
```
Comment (dilini-muthumala) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.is.analytics/widgets/IsAnalyticsHorizontalBarChart/src/IsAnalyticsHorizontalBarChart_bk.jsx]: Need license header.

File: components/sp-solutions/org.wso2.carbon.sp.solutions.is.analytics/widgets/IsAnalyticsHorizontalBarChart/src/pagination.jsx [hunk 5]
```diff
@@ -0,0 +1,192 @@
+import React from 'react';
```
Comment (dilini-muthumala) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.is.analytics/widgets/IsAnalyticsHorizontalBarChart/src/pagination.jsx]: Need license header.

File: components/sp-solutions/org.wso2.carbon.sp.solutions.is.analytics/widgets/IsAnalyticsHorizontalBarChart/webpack.config.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
+const path = require('path');
```
Comment (dilini-muthumala) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.is.analytics/widgets/IsAnalyticsHorizontalBarChart/webpack.config.js]: Need license header.

File: components/sp-solutions/org.wso2.carbon.sp.solutions.is.analytics/widgets/IsAnalyticsSummary/src/IsAnalyticsSummary.jsx [hunk 7]
```diff
@@ -0,0 +1,299 @@
...
+
+import Widget from '@wso2-dashboards/widget';
+import VizG from 'react-vizgrammar';
+import {MuiThemeProvider, createMuiTheme} from '@material-ui/core/styles';
+import _ from 'lodash';
+import Button from '@material-ui/core/Button';
+
+const widgetTexts = {
+    Overall: {
+        bodyText: 'Analyze overall login attempts made via WSO2 Identity Server. This includes information about overall flows of authentication took place through Identity Server. A collection of authentication steps is considered as an overall attempt',
```
Comment (dilini-muthumala) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.is.analytics/widgets/IsAnalyticsSummary/src/IsAnalyticsSummary.jsx]: Line length exceeds here.

File: features/sp-solution-features/org.wso2.carbon.sp.solutions.is.analytics.feature/pom.xml [hunk 8]
```diff
@@ -0,0 +1,190 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (dilini-muthumala) [inline features/sp-solution-features/org.wso2.carbon.sp.solutions.is.analytics.feature/pom.xml]: Need license header.

File: components/sp-solutions/org.wso2.carbon.sp.solutions.is.analytics/widgets/IsAnalyticsAreaChart/pom.xml [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+
+    <artifactId>org.wso2.sp.solutions.widgets.is.analytics.area.chart</artifactId>
+    <packaging>pom</packaging>
+    <name>WSO2 Stream Processor - Solutions - Widgets - IS Analytics - Area Chart</name>
+    <url>http://wso2.org</url>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.codehaus.mojo</groupId>
```
Comment (Niveathika92) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.is.analytics/widgets/IsAnalyticsAreaChart/pom.xml]: Move all plugin configs to the parent pom under plugin management. Then the plugin can be enabled in submodules with only the group id and name.

File: components/sp-solutions/org.wso2.carbon.sp.solutions.is.analytics/widgets/IsAnalyticsAreaChart/src/IsAnalyticsAreaChart.jsx [hunk 10]
```diff
@@ -0,0 +1,303 @@
...
+
+    componentWillUnmount() {
+        super.getWidgetChannelManager().unsubscribeWidget(this.props.id);
+    }
+
+    onReceivingMessage(message) {
+        if (message.header === "additionalFilterConditions") {
+            if (message.body === "") {
+                this.setState({
+                    additionalFilterConditions: undefined,
```
Comment (Niveathika92) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.is.analytics/widgets/IsAnalyticsAreaChart/src/IsAnalyticsAreaChart.jsx]: Do we have to intentionally set to undefined?

File: components/sp-solutions/org.wso2.carbon.sp.solutions.is.analytics/widgets/IsAnalyticsAreaChart/src/resources/widgetConf.json [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
...rocessedAuthenticationDataStream ( meta_tenantId int, eventId string, username string, localUsername string, userStoreDomain string, tenantDomain string, remoteIp string, region string, inboundAuthType string, serviceProvider string, rememberMeEnabled bool, forceAuthEnabled bool, passiveAuthEnabled bool, rolesCommaSeparated string, authenticationStep string, identityProvider string, authStepSuccess bool, stepAuthenticator string, isFirstLogin bool, successValue int, failureValue int, stepSuccessValue int, firstLoginValue int, identityProviderType string, timestamp long ); @store(type='rdbms', datasource='IS_ANALYTICS_DB') define aggregation AuthenticationStatAggregation from ProcessedAuthenticationDataStream select meta_tenantId, username, localUsername, rolesCommaSeparated, serviceProvider, identityProvider, remoteIp, region, userStoreDomain, isFirstLogin, sum(successValue) as authSuccessCount, sum(failureValue) as authFailureCount, sum(stepSuccessValue) as authStepSuccessCount, sum(firstLoginValue) as authFirstSuccessCount, identityProviderType, timestamp group by meta_tenantId, username, localUsername, remoteIp, rolesCommaSeparated aggregate by timestamp every second ... year;",
```
Comment (Niveathika92) [inline components/sp-solutions/org.wso2.carbon.sp.solutions.is.analytics/widgets/IsAnalyticsAreaChart/src/resources/widgetConf.json]: Remove all other processing logic here, except the aggregation and the stream.

File: features/is-analytics-solution-features/org.wso2.carbon.sp.solutions.is.analytics.feature/src/test/resources/testng.xml [hunk 12]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Niveathika92) [inline features/is-analytics-solution-features/org.wso2.carbon.sp.solutions.is.analytics.feature/src/test/resources/testng.xml]: Is there a need for testng.xml file inside features?

File: features/is-analytics-solution-features/org.wso2.carbon.sp.solutions.is.analytics.feature/pom.xml [hunk 13]
```diff
@@ -0,0 +1,205 @@
...
+            <groupId>org.wso2.analytics.solutions</groupId>
+            <artifactId>org.wso2.analytics.is.analytics.solutions.widgets.summary</artifactId>
+            <type>zip</type>
+        </dependency>
+        <dependency>
+            <groupId>org.wso2.analytics.solutions</groupId>
+            <artifactId>org.wso2.analytics.is.analytics.solutions.widgets.user.preferences</artifactId>
+            <type>zip</type>
+        </dependency>
+        <dependency>
```
Comment (Niveathika92) [inline features/is-analytics-solution-features/org.wso2.carbon.sp.solutions.is.analytics.feature/pom.xml]: Do we need a testng dependency in the feature?

File: components/sp-solutions/org.wso2.carbon.sp.solutions.is.analytics/pom.xml [hunk 14]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (ThisaruGuruge) [reply components/sp-solutions/org.wso2.carbon.sp.solutions.is.analytics/pom.xml]: Done

File: components/sp-solutions/org.wso2.carbon.sp.solutions.is.analytics/widgets/IsAnalyticsSummary/src/IsAnalyticsSummary.jsx (hunk 7 above)
Comment (ThisaruGuruge) [reply components/sp-solutions/org.wso2.carbon.sp.solutions.is.analytics/widgets/IsAnalyticsSummary/src/IsAnalyticsSummary.jsx]: Isn't it okay to exceed the line length for large strings ?

File: features/is-analytics-solution-features/org.wso2.carbon.sp.solutions.is.analytics.feature/pom.xml (hunk 13 above)
Comment (ThisaruGuruge) [reply features/is-analytics-solution-features/org.wso2.carbon.sp.solutions.is.analytics.feature/pom.xml]: No, will remove it

File: features/is-analytics-solution-features/org.wso2.carbon.sp.solutions.is.analytics.feature/src/test/resources/testng.xml (hunk 12 above)
Comment (ThisaruGuruge) [reply features/is-analytics-solution-features/org.wso2.carbon.sp.solutions.is.analytics.feature/src/test/resources/testng.xml]: No. Will remove it/

File: components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/IsAnalyticsAreaChart/src/IsAnalyticsAreaChart.jsx [hunk 15]
```diff
@@ -0,0 +1,309 @@
...
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+
```
Comment (Niveathika92) [inline components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/IsAnalyticsAreaChart/src/IsAnalyticsAreaChart.jsx]: React is not imported

File: components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/IsAnalyticsAreaChart/webpack.config.js [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+    },
+    plugins: [
+        new CopyWebpackPlugin([
+            {from: path.resolve(__dirname, './src/resources/')}
+        ])
+    ],
+    resolve: {
+        extensions: ['.js', '.json', '.jsx', '.scss']
+    }
+};
```
Comment (Niveathika92) [inline components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/IsAnalyticsAreaChart/webpack.config.js]: Refrain from packing react in each widget
https://github.com/wso2/analytics-solutions/blob/master/components/sp-solutions/org.wso2.carbon.sp.solutions.http.analytics/widgets/HTTPAnalyticsResponseCodeFilter/webpack.config.js#L83

File: components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/IsAnalyticsAttemptsOverTime/package.json [hunk 17]
```diff
@@ -0,0 +1,36 @@
+{
+  "name": "IsAnalyticsAttemptsOverTime",
+  "version": "1.0.0",
+  "private": true,
+  "dependencies": {
+    "@material-ui/core": "^1.4.0",
+    "@wso2-dashboards/widget": "^1.2.2",
+    "css-loader": "^0.28.11",
```
Comment (Niveathika92) [inline components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/IsAnalyticsAttemptsOverTime/package.json]: Are the loaders needed as direct dependencies? Cant we add them in dev dependency (If this is needed only as peer dependency)?

File: components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/IsAnalyticsAreaChart/webpack.config.js (hunk 16 above)
Comment (ThisaruGuruge) [reply components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/IsAnalyticsAreaChart/webpack.config.js]: Done. Added this line for each widget

File: components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/IsAnalyticsAreaChart/src/IsAnalyticsAreaChart.jsx (hunk 15 above)
Comment (ThisaruGuruge) [reply components/is-analytics-solutions/org.wso2.carbon.sp.is.analytics.solutions/widgets/IsAnalyticsAreaChart/src/IsAnalyticsAreaChart.jsx]: Done. Imported "react from React" in each widget

**Pull request merged**